draconic/interpreter.py

```diff
@@ -645,12 +702,49 @@ def _assign_unpack(self, names, values):
                 raise DraconicValueError(
                     f"Cannot unpack non-iterable {type(values).__name__} object", names, self._expr
                 )
-            if not len(names.elts) == len(values):
-                raise DraconicValueError(
-                    f"Unequal unpack: {len(names.elts)} names, {len(values)} values", names, self._expr
-                )
-            for t, v in zip(names.elts, values):
-                self._assign_unpack(t, v)
+
+            stars = (i for i in names.elts if type(i) is ast.Starred)
+            starred = next(stars, None)
+
+            if starred is None:
+                if len(names.elts) > len(values):
+                    raise DraconicValueError(
+                        f"not enough values to unpack (expected {len(names.elts)}, got {len(values)})",
+                        values,
+                        self._expr,
+                    )
+                elif len(names.elts) < len(values):
+                    raise DraconicValueError(
+                        f"too many values to unpack (expected {len(names.elts)}, got {len(values)})",
+                        values,
+                        self._expr,
+                    )
+                for t, v in zip(names.elts, values):
+                    self._assign_unpack(t, v)
+            elif (extra := next(stars, None)) is not None:
+                raise DraconicSyntaxError.from_node(extra, "multiple starred expressions in assignment", self._expr)
+            else:
+                if len(values) < (len(names.elts) - 1):
+                    raise DraconicValueError(
+                        f"not enough values to unpack (expected at least {len(names.elts) - 1}, got {len(values)})",
+                        values,
+                        self._expr,
+                    )
+                for t, v in zip(names.elts, values):
+                    if t is starred:
+                        break
+                    values.remove(v)
+                    self._assign_unpack(t, v)
+
+                for t, v in zip(reversed(names.elts), reversed(values)):
+                    if t is starred:
+                        break
+                    values.remove(v)
+                    self._assign_unpack(t, v)
+                self._assign_unpack(starred, values)
```

Comment:
There's a util method to zip with respect to stars that might be useful here

Reply:
Apparently I just never noticed that, that would be much better.

Reply:
What is it called?

Reply:
Nevermind, I found it, the helper called `zip_star`, correct?

Reply:
Oh, it's in `helpers` instead of `utils` for some reason - `helpers.zip_star`